src/uu/dmesg/src/dmesg.rs

```diff
@@ -43,7 +50,71 @@ pub fn uumain(args: impl uucore::Args) -> UResult<()> {
             }
         };
     }
-    dmesg.parse()?.print();
+    if let Some(list_args) = matches.get_many::<String>(options::FACILITY) {
+        let mut facility_filters = HashSet::new();
+        for list in list_args {
+            for arg in list.split(',') {
+                let facility = match arg {
+                    "kern" => Facility::Kern,
+                    "user" => Facility::User,
+                    "mail" => Facility::Mail,
+                    "daemon" => Facility::Daemon,
+                    "auth" => Facility::Auth,
+                    "syslog" => Facility::Syslog,
+                    "lpr" => Facility::Lpr,
+                    "news" => Facility::News,
+                    "uucp" => Facility::Uucp,
+                    "cron" => Facility::Cron,
+                    "authpriv" => Facility::Authpriv,
+                    "ftp" => Facility::Ftp,
+                    "res0" => Facility::Res0,
+                    "res1" => Facility::Res1,
+                    "res2" => Facility::Res2,
+                    "res3" => Facility::Res3,
+                    "local0" => Facility::Local0,
+                    "local1" => Facility::Local1,
+                    "local2" => Facility::Local2,
+                    "local3" => Facility::Local3,
+                    "local4" => Facility::Local4,
+                    "local5" => Facility::Local5,
+                    "local6" => Facility::Local6,
+                    "local7" => Facility::Local7,
+                    _ => return Err(USimpleError::new(1, format!("unknown facility '{arg}'"))),
+                };
+                facility_filters.insert(facility);
+            }
+        }
+        dmesg.facility_filters = Some(facility_filters);
+    }
+    if let Some(list_args) = matches.get_many::<String>(options::LEVEL) {
+        let mut level_filters = HashSet::new();
+        for list in list_args {
+            for arg in list.split(',') {
+                let level = match arg {
+                    "emerg" => Level::Emerg,
+                    "alert" => Level::Alert,
+                    "crit" => Level::Crit,
+                    "err" => Level::Err,
+                    "warn" => Level::Warn,
+                    "notice" => Level::Notice,
+                    "info" => Level::Info,
+                    "debug" => Level::Debug,
+                    _ => return Err(USimpleError::new(1, format!("unknown level '{arg}'"))),
+                };
+                level_filters.insert(level);
+            }
+        }
+        dmesg.level_filters = Some(level_filters);
+    }
+    if let Some(since) = matches.get_one::<String>(options::SINCE) {
+        let since = remove_enclosing_quotes(since);
+        dmesg.since_filter = Some(time_formatter::parse_datetime(since)?);
+    }
+    if let Some(until) = matches.get_one::<String>(options::UNTIL) {
+        let until = remove_enclosing_quotes(until);
```

Comment:
What's the reason for calling `remove_enclosing_quotes`? Somehow I don't understand why a user would enclose the input twice with quotes. And the original `dmesg` fails with an "invalid time value" error if I provide additional quotes like `--since="\"2024-01-24\""`. Maybe I'm missing something?

Reply:
TL;DR: I misunderstood how `new_ucmd().arg()` handles its argument, and I'll push a commit to remove the function. Thanks! 😊

I just tried things out in I just realized that I misunderstood how `new_ucmd().arg()` handles the argument supplied.

So if you look at the test I wrote, I supply the argument with a string value of `--since=\"2024-11-19 17:47:32 +0700\"`, and so on. And that test will fail if remove the `remove_enclosing_quotes`.

So as you see, I didn't intend to support inputs that is enclosed by quotes twice. I added one enclosing quotes around the date because I thought `new_ucmd().arg()` cannot parse the argument correctly if I remove the enclosing quotes since it has a SPACE. But it turns out I was wrong, I should remove the quotes. And if I supply the argument through command line, it will remove the quotes automatically.

P.S. I tried this with the original code: `cargo run dmesg -K tests/fixtures/dmesg/kmsg.input --since="\"2024-12-13 15:02
:04\""` and actually it works. Nevertheless, it shouldn't work 😄

Reply:
Fixed in d5b2f3c